README.md

```diff
@@ -197,6 +189,29 @@ person.morning.value = "blue"                        # => SET people:5:morning
 true == person.morning.blue?                         # => GET people:5:morning
 ```
 
+### Default values
+
+You can set a default value for all types. For example:
+
+```ruby
+list = Kredis.list "favorite_colors", default: [ "red", "green", "blue" ]
+
+# or, in a model
+class Person < ApplicationRecord
+  kredis_string :name, default: "Unknown"
+  kredis_list :favorite_colors, default: [ "red", "green", "blue" ]
+end
+```
+
+There's a performance overhead to consider though. When you first read or write an attribute in a model, Kredis will
+check if the underlying Redis key exists, while watching for concurrent changes, and if it does not,
+write the specified default value.
+
+This means that using default values in a typical Rails app additional Redis calls (WATCH, EXISTS, UNWATCH) will be
+executed for each Kredis attribute with a default value read or written during a request.
```

Comment:
Does this occur every request, or only when a value does not exist?

Reply:
In the context of a Rails web request, as the Kredis object is initialized, the key is checked with WATCH, EXISTS, UNWATCH once, but then subsequent operations within the request do not incur any Redis command overhead. If no default value is specified, no additional Redis command overhead is incurred at all.

Reply:
### Scenario 1: Default NOT defined
(no additional overhead)

### Scenario 2: Default defined, value NOT exists
WATCH, EXISTS, write default (depending on type), UNWATCH

### Scenario 3: Default defined, value exists
WATCH, EXISTS, UNWATCH

Reply:
It seems a more efficient approach would be to only add extra overhead of default is defined AND value doesn't exist. In other words:

1. Get Value
2. Value doesn't exist
3. Execute Default Logic

This avoids extra overhead if the value has been set already.

Reply:
I agree @tleish. That's exactly scenario 3, but with the added WATCH / UNWATCH.

To illustrate the problem that the WATCH / UNWATCH solves check this test case:

https://github.com/rails/kredis/blob/b1a818ad3e70a23154ed90105a6f1d6f060cbbd1/test/types/counter_test.rb#L113-L121

Without the WATCH / UNWATCH, the value existing check is prone to a race condition.